src/app/app.functions/crm-card.js

```diff
@@ -1,74 +1,68 @@
-  // Static CRM card example
-  const staticCard = {
-    objectId: 1,
-    title: 'Sample project custom CRM card',
-    desc: `A custom CRM card is a UI extension that displays custom info from either HubSpot or an external system. It can also include custom actions—click the "Get Inspired" button to see example data from the ZenQuotes API.`,
-  };
 
-  // Make a call to zenquote public api and create a CRM card with result
-  const getExternalCard = async () => {
-    try {
-      const { data } = await axios.get('https://zenquotes.io/api/random');
+  // Store contact firstname, configured as propertiesToSend in crm-card.json
+  const { firstname } = context.propertiesToSend;
 
-      return {
-        objectId: 2,
-        link: 'https://zenquotes.io/',
-        title: 'Inspirational quotes provided by ZenQuotes API',
-        quote: data[0].q,
-        author: data[0].a,
-      };
-    } catch (error) {
-      throw new Error(
-        `There was an error fetching the quote': ${error.message}`
-      );
-    }
+  const introMessage = {
+    type: "text",
+    format: "markdown",
+    text: "_An example of a CRM card extension that displays data from Hubspot, uses ZenQuotes public API to display daily quote, and demonstrates custom actions using serverless functions._",
   };
 
-  // Make a call to the HubSpot contacts API and create a CRM card with result
-  const getInternalCard = async () => {
-    try {
-      // Fetching contacts from CRM
-      const resp = await hubspotClient.crm.contacts.basicApi.getPage();
+  let quoteSections = [];
 
-      return {
-        objectId: 3,
-        title: 'First contact in account',
-        firstname: resp.body.results[0].properties.firstname,
-      };
-    } catch (error) {
-      throw new Error(
-        `There was an error fetching the contact': ${error.message}`
-      );
-    }
-  };
-
-  // Assemble cards
   try {
-    const internalCard = await getInternalCard();
-    const externalCard = await getExternalCard();
+    const { data } = await axios.get("https://zenquotes.io/api/random");
 
-    sendResponse({
-      results: [staticCard, internalCard, externalCard],
-      primaryAction: {
-        type: 'SERVERLESS_ACTION_HOOK',
-        serverlessFunction: 'crm-card',
-        label: 'Get Inspired',
+    quoteSections = [
+      {
+        type: "tile",
+        body: [
+          {
+            type: "text",
+            format: "markdown",
+            text: "**Hello " + firstname + ", here's your quote for the day**!"
+          },
+          {
+            type: "text",
+            format: "markdown",
+            text: "**Quote**: " + data[0].q
+          },
+          {
+            type: "text",
+            format: "markdown",
+            text: "**Author**: " + data[0].a
+          }
+        ]
       },
-    });
+      {
+        type: "button",
+        text: "Get new quote",
+        onClick: {
+          type: "SERVERLESS_ACTION_HOOK",
+          serverlessFunction: "crm-card"
```

Comment:
Will this work? Can the action button callback be the same as the card's primary target function?

Even if it will work, this might be a good opportunity to illustrate having multiple serverless functions in one project.

Reply:
We should continue this conversation, but I'm going to get this PR merged as-is for the empathy session.